File: app/src/main/java/com/rick/javavskotlinfortuan/java/spec/Reportable.java
package com.rick.javavskotlinfortuan.java.spec;

import android.util.Log;

public interface Reportable {

    String TAG = Reportable.class.getSimpleName();

    default void report(String info) {
        // do report...
        Log.d(TAG, "report: " + info);
    }
}

File: app/src/main/java/com/rick/javavskotlinfortuan/kotlin/spec/Reportable.kt
package com.rick.javavskotlinfortuan.kotlin.spec

import android.util.Log

interface Reportable {

    companion object {
        val TAG = Reportable::class.simpleName
    }

    fun report(info: String?) {
        // do report...
        Log.d(TAG, "report: $info")
    }
}
